util/generate_haddock3_catalog.py

```diff
@@ -135,48 +222,112 @@ def config2schema(config):
+                    if obj_schemas['uiSchema']:
+                        prop_ui = {
+                            "items": obj_schemas['uiSchema']
+                        }
+                    prop_toml = {
+                        'indexed': True,
+                        'items': {
+                            'flatten': True
+                        }
+                    }
+                elif v['dim'] == 2:
+                    prop['items'] = {
+                        'type': 'array',
+                        'items': obj_schemas['schema']
                     }
+                    if obj_schemas['uiSchema']:
+                        prop_ui = {
+                            "items": {
+                                'items': obj_schemas['uiSchema']
+                            }
+                        }
+                    prop_toml = {
+                        'indexed': True,
+                        'items': {
+                            'indexed': True,
+                            'items': {
+                                'flatten': True
+                            }
+                        }
+                    }
+                else:
+                    raise Exception('Unknown dim')
             elif 'items' in v:
-                obj = {'a' : v['items']} # config2schema requires object
-                schema_uiSchema = config2schema(obj)
-                prop['items'] = schema_uiSchema['schema']['properties']['a']
-                if schema_uiSchema['uiSchema'] and schema_uiSchema['uiSchema']['a']:
-                    prop_ui = {
-                        "items": schema_uiSchema['uiSchema']['a']
+                obj_schemas = config2schema({'a': v['items']})
+                if v['dim'] == 1:
+                    prop['items'] = obj_schemas['schema']['properties']['a']
+                    if 'a' in obj_schemas['uiSchema']:
+                        prop_ui = {
+                            "items": obj_schemas['uiSchema']['a']
+                        }
+                    prop_toml = {
+                        'indexed': True
+                    }
+                elif v['dim'] == 2:
+                    prop['items'] = {
+                        'type': 'array',
+                        'items': obj_schemas['schema']['properties']['a']
                     }
-            elif len(v['default']) and (isinstance(v['default'][0], int) or isinstance(v['default'][0], float)):
+                    if 'a' in obj_schemas['uiSchema']:
+                        prop_ui = {
+                            "items": {
+                                'items': obj_schemas['uiSchema']['a']
+                            }
+                        }
+                    prop_toml = {
+                        'indexed': True,
+                        'items': {
+                            'indexed': True
+                        }
+                    }
+                else:
+                    raise Exception('Unknown dim')
```

Comment:
Can this error message be clarified? 
```suggestion
                    raise Exception(f'Invalid value of dim in {v=}')
```